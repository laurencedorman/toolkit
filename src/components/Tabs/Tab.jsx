/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';

const Tab = ({ children, label }) => <div label={label}>{children}</div>;

Tab.displayName = 'Tab';

Tab.propTypes = {
  children: PropTypes.node.isRequired,
  label: PropTypes.node.isRequired,
};

export default Tab;
