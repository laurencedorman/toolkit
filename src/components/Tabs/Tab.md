Tab:

- Put Tab component as child of Tabs component.
- Label props define automatically TabLabel selector.

```js
<Tab label="tabOne">
  <h3>Tab content</h3>
  <p>i work with Tabs component</p>
</Tab>
```
