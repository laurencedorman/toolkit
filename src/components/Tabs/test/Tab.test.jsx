/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow } from 'enzyme';
import Tab from '../Tab';

describe('Tab', () => {
  it('should match default snapshot', () => {
    const wrapper = shallow(<Tab label="label" isActiveTab />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match children snapshot', () => {
    const wrapper = shallow(
      <Tab label="label">
        <span>This is a children</span>
      </Tab>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
