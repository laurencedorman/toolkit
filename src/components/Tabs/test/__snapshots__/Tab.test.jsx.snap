// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should match children snapshot 1`] = `
<div
  label="label"
>
  <span>
    This is a children
  </span>
</div>
`;

exports[`Tab should match default snapshot 1`] = `
<div
  label="label"
/>
`;
